### PR TITLE
feature(lightweight transaction): new longevity with big data set

### PR DESCRIPTION
**Pull request by juliayakovlev**
1. New LWT logevity with bigger data set 500GB.
It's planned to be run for 3 days. For now I defined it
temporary for 15 hours. It still under developing.

2. Remove running repair before data validation because of
it's not needed

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: data_dir/c-s_lwt_big_data.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+# The CQL for creating a table you wish to stress (optional if it already exists)
+table_definition: |
+  CREATE TABLE IF NOT EXISTS blogposts (
+        domain int,
+        published_date int,
+        lwt_indicator int,
+        url blob,
+        url1 blob,
+        url2 blob,
```
Comment (roydahan) [inline data_dir/c-s_lwt_big_data.yaml:22]: Why only blobs?
Let's try to have it cover also other types? even many types.

File: data_dir/c-s_lwt_big_data.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  - create MATERIALIZED VIEW blogposts_update_two_columns_lwt_indicator as select lwt_indicator, author from blogposts where domain is not null and published_date is not null and lwt_indicator > 0 and lwt_indicator <= 1000000 PRIMARY KEY(lwt_indicator, domain, published_date);
+  - create MATERIALIZED VIEW blogposts_update_two_columns_lwt_indicator_upd as select lwt_indicator, author from blogposts where domain is not null and published_date is not null and lwt_indicator = 20000000 PRIMARY KEY(lwt_indicator, domain, published_date);
+  - create MATERIALIZED VIEW blogposts_not_updated_lwt_indicator as select lwt_indicator, author from blogposts where domain is not null and published_date is not null and lwt_indicator > 1005000 and lwt_indicator < 20000000 PRIMARY KEY(lwt_indicator, domain, published_date);
+
+### Column Distribution Specifications ###
+
+columnspec:
+  - name: domain
+    population: seq(1..500000000)  #10M possible domains to pick from
```
Comment (roydahan) [inline data_dir/c-s_lwt_big_data.yaml:42]: fix comment

File: data_dir/c-s_lwt_big_data.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  - name: published_date
+    cluster: uniform(1..2000)         #under each domain we will have max 1000 posts
```
Comment (roydahan) [inline data_dir/c-s_lwt_big_data.yaml:45]: fix comment

File: data_dir/c-s_lwt_big_data.yaml (hunk 1 above)
Comment (roydahan) [reply data_dir/c-s_lwt_big_data.yaml:22]: it's ok if it will change the size to be more than 500GB

File: data_dir/c-s_lwt_big_data.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  - name: url2
+    size: fixed(200)
+
+  - name: url3
+    size: fixed(200)
+
+  - name: url4
+    size: fixed(200)
+
+  - name: title                  #titles shouldn't go beyond 200 chars
```
Comment (roydahan) [inline data_dir/c-s_lwt_big_data.yaml:65]: fix comment

File: data_dir/c-s_lwt_big_data.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+  - name: author
+    size: uniform(5..20)         #author names should be short
+
+### Batch Ratio Distribution Specifications ###
+
+insert:
+  partitions: fixed(1)            # Our partition key is the domain so only insert one per batch
+
+  select:    fixed(1)/1000        # We have 1000 posts per domain so 1/1000 will allow 1 post per batch
```
Comment (roydahan) [inline data_dir/c-s_lwt_big_data.yaml:76]: Now we have 2000, don't we need to change it?

File: longevity_lwt_test.py [hunk 6]
```diff
@@ -211,33 +212,13 @@ def start_nemesis(self):
...
-                            table_name='blogposts_not_updated_lwt_indicator')
-
     def test_lwt_longevity(self):
         self.test_custom_time()
 
-        # Run repair on all nodes. We need it to fix the case, when part of data wasn't inserted because of timeouts
-        self.stop_nemesis_and_repair_cluster()
+        # Stop nemesis. Prefer all nodes will be run before data validation
+        if self.db_cluster.nemesis_threads:
+            self.db_cluster.stop_nemesis()
```
Comment (roydahan) [inline longevity_lwt_test.py:220]: When you stop_nemesis does it wait for the nemesis thread to stop and the actual nemesis that runs to finish?

File: test-cases/longevity/longevity-lwt-500G-3d.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# Test duration 15 hours temporary. Should be 3 days. Will be changed later
+test_duration: 900
+prepare_write_cmd: ["cassandra-stress user profile=/tmp/c-s_lwt_big_data.yaml n=400000000 ops'(insert=1)' cl=QUORUM -mode native cql3 -rate threads=50" ]
```
Comment (roydahan) [inline test-cases/longevity/longevity-lwt-500G-3d.yaml:3]: How long does it take to populate?
What is the load during populate?

File: test-cases/longevity/longevity-lwt-500G-3d.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+stress_cmd: ["cassandra-stress user profile=/tmp/c-s_lwt_big_data.yaml ops'(lwt_update_one_column=1)' cl=QUORUM duration=180m -mode native cql3 -rate threads=30" ,
+             "cassandra-stress user profile=/tmp/c-s_lwt_big_data.yaml ops'(lwt_update_two_columns=1)' cl=QUORUM duration=180m -mode native cql3 -rate threads=30"
+            ]
+stress_read_cmd: ["cassandra-stress user profile=/tmp/c-s_lwt_big_data.yaml ops'(select=1)' cl=SERIAL duration=180m -mode native cql3 -rate threads=20" ]
+
+n_db_nodes: 4
+n_loaders: 3
+n_monitor_nodes: 1
+round_robin: true
```
Comment (roydahan) [inline test-cases/longevity/longevity-lwt-500G-3d.yaml:12]: With round_robin I think the 3rd loader won't do anything with this configuration

File: test-cases/longevity/longevity-lwt-500G-3d.yaml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+instance_type_db: 'i3.2xlarge'
```
Comment (roydahan) [inline test-cases/longevity/longevity-lwt-500G-3d.yaml:14]: Maybe we should consider changing this to i3.4x

File: longevity_lwt_test.py (hunk 6 above)
Comment (juliayakovlev) [reply longevity_lwt_test.py:220]: Yes, it joins the nemesis thread (with timeout).

File: test-cases/longevity/longevity-lwt-500G-3d.yaml (hunk 7 above)
Comment (juliayakovlev) [reply test-cases/longevity/longevity-lwt-500G-3d.yaml:3]: It's pretty long. I am working on it to improve the time

File: test-cases/longevity/longevity-lwt-500G-3d.yaml (hunk 8 above)
Comment (juliayakovlev) [reply test-cases/longevity/longevity-lwt-500G-3d.yaml:12]: Why?

File: data_dir/c-s_lwt_big_data.yaml (hunk 2 above)
Comment (juliayakovlev) [reply data_dir/c-s_lwt_big_data.yaml:42]: fixed

File: data_dir/c-s_lwt_big_data.yaml (hunk 3 above)
Comment (juliayakovlev) [reply data_dir/c-s_lwt_big_data.yaml:45]: fixed

File: data_dir/c-s_lwt_big_data.yaml (hunk 4 above)
Comment (juliayakovlev) [reply data_dir/c-s_lwt_big_data.yaml:65]: fixed

File: data_dir/c-s_lwt_big_data.yaml (hunk 5 above)
Comment (juliayakovlev) [reply data_dir/c-s_lwt_big_data.yaml:76]: fixed

File: test-cases/longevity/longevity-lwt-500G-3d.yaml (hunk 9 above)
Comment (juliayakovlev) [reply test-cases/longevity/longevity-lwt-500G-3d.yaml:14]: Do we need that? There is 1.5 T space on the disk

File: data_dir/c-s_lwt_big_data.yaml (hunk 1 above)
Comment (juliayakovlev) [reply data_dir/c-s_lwt_big_data.yaml:22]: Done

File: test-cases/longevity/longevity-lwt-500G-3d.yaml (hunk 7 above)
Comment (juliayakovlev) [reply test-cases/longevity/longevity-lwt-500G-3d.yaml:3]: With i3.4xlarge db node and c5.2xlarge loaders it takes about 9 hours

**Comment by juliayakovlev** [comment]
Closed because of #2102  includes all features from this PR

**Pull request closed without merging**